### PR TITLE
Initiated refactor to split serialization/deserialization logic from mainwindow

### PR DESCRIPTION
**Pull request by stefanodellosa-personal**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,6 +35,9 @@ build/
 *.cache
 *.sublime-workspace
 
+# Env
+.conda/
+
```
Comment (alessandropalla) [inline .gitignore:40]: is this ok to be tracked?

File: wadas/domain/animal_detection_mode.py [hunk 2]
```diff
@@ -16,6 +16,11 @@ def __init__(self, classification=True):
         super(AnimalDetectionAndClassificationMode, self).__init__()
         self.process_queue = True
         self.en_classification = classification
+        self.type = (
+            OperationMode.OperationModeTypes.AnimalDetectionAndClassificationMode
+            if classification
+            else OperationMode.OperationModeTypes.AnimalDetectionMode
+        )
```
Comment (alessandropalla) [inline wadas/domain/animal_detection_mode.py:23]: you can rewrite by not using a ternary operation but a normal if/else clause that won't overflow the max line

File: wadas/domain/configuration.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+from wadas.domain.feeder_actuator import FeederActuator
+from wadas.domain.ftp_camera import FTPCamera
+from wadas.domain.ftps_server import FTPsServer
+from wadas.domain.notifier import Notifier
+from wadas.domain.operation_mode import OperationMode
+from wadas.domain.roadsign_actuator import RoadSignActuator
+from wadas.domain.usb_camera import USBCamera
+
+logger = logging.getLogger(__name__)
+WADAS_VERSION = 0.1
```
Comment (alessandropalla) [inline wadas/domain/configuration.py:23]: TODO: use a better way to track system version. In particular we should have a unique point in the packet for this. The standard way is to have a `__version__` variable in the module `__init__.py` file and then you get it by using `wadas.__version__`

File: wadas/domain/configuration.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+                        FTPsServer.ftps_server.add_user(
+                            credentials.username,
+                            credentials.password,
+                            ftp_camera.ftp_folder,
+                        )
+        Camera.detection_params = wadas_config["camera_detection_params"]
+        FastAPIActuatorServer.actuator_server = (
+            FastAPIActuatorServer.deserialize(wadas_config["actuator_server"])
+            if wadas_config["actuator_server"]
+            else None
```
Comment (alessandropalla) [inline wadas/domain/configuration.py:77]: like above

File: wadas/domain/configuration.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        )
+        AiModel.detection_treshold = wadas_config["ai_model"]["ai_detect_treshold"]
+        AiModel.classification_treshold = wadas_config["ai_model"]["ai_class_treshold"]
+        AiModel.language = wadas_config["ai_model"]["ai_language"]
+        selected_operation_mode = (
+            OperationMode.OperationModeTypes(wadas_config["operation_mode"])
+            if wadas_config["operation_mode"]
+            else None
```
Comment (alessandropalla) [inline wadas/domain/configuration.py:85]: the same

File: .gitignore (hunk 1 above)
Comment (stefanodellosa-personal) [reply .gitignore:40]: will remove it in next commit

**Review by alessandropalla — APPROVED** [review]

**Pull request merged**